sdk/identity/src/token_credentials/auto_refreshing_credentials.rs

```diff
@@ -0,0 +1,66 @@
+use async_lock::RwLock;
+use azure_core::{
+    auth::{TokenCredential, TokenResponse},
+    Error,
+};
+use chrono::{Duration, Utc};
+use std::sync::Arc;
+
+fn is_expired(token: TokenResponse) -> bool {
```

Comment:
Can this just take a `&TokenResponse` so you don't have to clone it all the time?